### DISCUSSION TITLE
使用QCefView打开网页，无法从网页下载任何东西

### DISCUSSION THREAD

**Discussion by L-ZT** [CefView/QCefView#169] [Q&A] [answered]
使用QCefView的demo，无法从打开的网页中下载任何东西，大家遇见过吗？请问如何解决？原生cef的demo是可以的。

**Comment by tishion** [comment, marked as answer]
demo已经实现了下载功能的，不知道你说的不能下载是什么意思。

**Reply by L-ZT** [reply]
我试着用demo的下载功能，好像也是不行。这里的不能下载指的是打开文件地址无法下载。

**Reply by L-ZT** [reply]
https://nodejs.org/en/在这个网站上，点击下载的时候不起作用

**Reply by L-ZT** [reply]
你好，这个问题能帮忙抽空看下吗？谢谢。

**Reply by tishion** [reply]
下载功能是有的，现在也可以凑合着用，不过cef在100的时候又更新了，所以需要再修一下。
demo里面有代码，你自己看看吧。

**Reply by L-ZT** [reply]
抱歉，打扰一下。我已经修改了demo的代码，并将右边的窗口设置的url设置成www.baidu.com，然后在上面随便下载某个文件（这里我下载了notepad++这个安装包），但是没有任何反应。请你有空帮忙看下QCefView是否需要做一些特色设置。
![image](https://user-images.githubusercontent.com/65488449/216605696-4e054984-7d62-42d8-b5bc-0eab05a91a55.png)
![image](https://user-images.githubusercontent.com/65488449/216605928-573c9541-b770-4f10-8e9a-c834d7d683e8.png)

**Reply by L-ZT** [reply]
![image](https://user-images.githubusercontent.com/65488449/216609491-a37b7e84-044a-4c7c-a4a0-8f6fea3a26d8.png)

**Reply by tishion** [reply]
下载的相关的两个处理函数是跟QCefView的实例相关的，弹出的新窗口不是QCefView的实例，你就没有机会去实现那两个处理函数，当然没有反应。直接把左边的QCefView设置成包含你的下载页面的地址就行了。
